src/main/java/de/eddyson/tapestry/react/services/nashorn/NashornReactRenderEngineImplementation.java

```diff
@@ -0,0 +1,140 @@
+package de.eddyson.tapestry.react.services.nashorn;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.util.Map;
+
+import javax.script.Bindings;
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+import javax.script.SimpleBindings;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.tapestry5.annotations.Path;
+import org.apache.tapestry5.internal.services.assets.ResourceChangeTracker;
+import org.apache.tapestry5.ioc.Resource;
+import org.apache.tapestry5.ioc.annotations.Inject;
+import org.apache.tapestry5.json.JSONObject;
+import org.apache.tapestry5.services.assets.StreamableResourceSource;
+import org.apache.tapestry5.services.javascript.ModuleManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Stopwatch;
+import com.google.common.collect.Maps;
+
+public class NashornReactRenderEngineImplementation implements ReactRenderEngine {
+
+  private static final Logger log = LoggerFactory.getLogger(NashornReactRenderEngineImplementation.class);
+
+  private final StreamableResourceSource srs;
+  private final ResourceChangeTracker tracker;
+  private final ModuleManager moduleManager;
+  private final Resource rJsResource;
+  
+  private ScriptEngine _engine;
+  private SimpleBindings _bindings;
+
+
+  public NashornReactRenderEngineImplementation(@Inject @Path("webjars:requirejs-node:$version/r.js") Resource rJsResource, ModuleManager moduleManager, StreamableResourceSource srs,
+      ResourceChangeTracker tracker) {
+    super();
+    this.rJsResource = rJsResource;
+    this.moduleManager = moduleManager;
+    this.srs = srs;
+    this.tracker = tracker;
+  }
+
+  @Override
+  public String renderReactComponent(String moduleName, JSONObject parameters) {
+    Stopwatch watch = Stopwatch.createStarted();
+    try {
+      StringBuilder builder = new StringBuilder();
+      builder.append("var reactResult = null;\n");
+      builder.append(
+          "require(['eddyson/react/isomorphic/run', '%1$s'], function(ri) { reactResult = ri('%1$s', %2$s); } );\n");
+      builder.append("reactResult;");
+      String script = String.format(builder.toString(), moduleName, parameters.toCompactString());
```

Comment:
Can you avoid this `String.format` call`? It's quite expensive and since you're using a `StringBuilder` anyway...